From 1cee10b74621b8d70ff31b6e358ee52bd57e695e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B3zef=20Kucia?= <jkucia@codeweavers.com>
Date: Sat, 2 Jan 2016 11:48:45 +0100
Subject: [PATCH] ddraw: Prevent division by zero in viewport_activate.
Reply-To: wine-devel <wine-devel@winehq.org>

This just stop the crash, without fixing the actual cause.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=36763
---
 dlls/ddraw/viewport.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlls/ddraw/viewport.c b/dlls/ddraw/viewport.c
index 5ebc41e..16cdad4 100644
--- a/dlls/ddraw/viewport.c
+++ b/dlls/ddraw/viewport.c
@@ -71,6 +71,10 @@ void viewport_activate(struct d3d_viewport *This, BOOL ignore_lights)
     /* And copy the values in the structure used by the device */
     if (This->use_vp2)
     {
+        if (!This->viewports.vp2.dvMinZ && !This->viewports.vp2.dvMaxZ
+                && !This->viewports.vp2.dvClipWidth && !This->viewports.vp2.dvClipHeight)
+            return;
+
         vp.dwX = This->viewports.vp2.dwX;
         vp.dwY = This->viewports.vp2.dwY;
         vp.dwHeight = This->viewports.vp2.dwHeight;
-- 
2.4.10

